OutboxSample/Infrastructure/UnitOfWorkFactory.cs

```diff
@@ -179,9 +179,13 @@ private void EnsureHasBeenRequestedToOpen()
     }
 }
 
-internal class DbCommandProxy : IDbCommand
+internal interface IDbCommandProxy
```

Comment:
It doesn't look to me that we really need this interface. Since the `DbCommandProxy` is mean to be just an implementation detail, we don't need to bother to put another abstraction layer (i.e. the interface) on top of it. Also, it's highly unlikely that there'll ever be another implementation of this interface.
Imo just `DbCommandProxy` should be enough. For convenience, we might want to expose its `LiveCommand` property as `internal`.